mesonbuild/optinterpreter.py

```diff
@@ -145,17 +170,24 @@ def evaluate_statement(self, node: mparser.BaseNode) -> None:
         (posargs, kwargs) = self.reduce_arguments(node.args)
         self.func_option(posargs, kwargs)
 
-    @typed_kwargs('option',
-                  KwargInfo('type', str, required=True),
-                  KwargInfo('description', str, default=''),
-                  KwargInfo('yield', bool, default=coredata.default_yielding, since='0.45.0'),
-                  KwargInfo('choices', (ContainerTypeInfo(list, str), type(None))),
-                  KwargInfo('value', object),
-                  KwargInfo('min', (int, type(None))),
-                  KwargInfo('max', (int, type(None))),
-                  KwargInfo('deprecated', (bool, str, ContainerTypeInfo(dict, str), ContainerTypeInfo(list, str)),
-                            default=False, since='0.60.0')
-                  )
+    @typed_kwargs(
+        'option',
+        KwargInfo(
+            'type',
+            str,
+            required=True,
+            validator=in_set_validator({'string', 'boolean', 'integer', 'combo', 'array', 'feature'})
+        ),
+        KwargInfo('description', str, default=''),
+        KwargInfo(
+            'deprecated',
+            (bool, str, ContainerTypeInfo(dict, str), ContainerTypeInfo(list, str)),
+            default=False,
+            since='0.60.0',
+            feature_validator=lambda x: [FeatureNew('string value to "deprecated" keyword argument', '0.63.0')] if isinstance(x, str) else []
```

Comment:
I think that can be done with `since_value={str: '0.63'}`, right?

Reply:
nope, since value is not for types its for values. so you could do something like:
```meson
since_values={'foo': '1.1.')
```
Honestly, given that we know have feature_validator I'm very tempted to just delete since_values ans provide some helpers for doing what since_values and deprecated_values do, since the code is actually already split up like that

Reply:
c1aeb2f7e4ef81a046ca28d5dad89bb8fdeefafd

> typed_kwargs: support dict/list as "new since" types

In retrospect it was a bit odd to implement this as `if n in {dict, list}:` instead of `if isinstance(n, type)` to check whether since_values is getting a type object instead of an object *instance*.

As it is, we cannot use this for str.

Reply:
I thought https://github.com/mesonbuild/meson/pull/11061 was merged already, there I extend to allow any type: https://github.com/mesonbuild/meson/pull/11061/commits/88166de9859b8025a2b57bf27d8f9979538573e8. It already can be dict or list types. Definitely easier than this lambda, but with some helpers I'm not against it.